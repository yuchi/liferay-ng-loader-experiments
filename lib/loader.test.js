
const glob = require('glob');
const path = require('path');
const fs = require('fs');

const concat = (a, b) => a.concat(b);

const packages = glob.sync('fixtures/*/*').map(pathname => {
    const pkg = JSON.parse(fs.readFileSync(
        path.join(pathname, 'package.json'),
        'utf8')
    );

    return Object.assign({
        dependencies: {},
        peerDependencies: {},
        modules: glob.sync(path.join(pathname, '**', '*')).map(filename => ({
            filename,
            local: path.relative(pathname, filename),
            content: fs.readFileSync(filename, 'utf8')
        }))
    }, pkg);
});

const Loader = require('./loader');

const loader = new Loader(packages);

test('Should work', () => {
    expect(loader.execute('app1@1.0.0', 'index.js')).toMatchSnapshot();
});

test('Should work', () => {
    expect(loader.execute('app2@1.0.0', 'index.js')).toMatchSnapshot();
});

test('Should work', () => {
    expect(loader.execute('app3@1.0.0', 'index.js')).toMatchSnapshot();
});

test('Should throw with circular peerDependencies', () => {
    expect(() => {
        loader.execute('app4@1.0.0', 'index.js');
    }).toThrowErrorMatchingSnapshot();
});
