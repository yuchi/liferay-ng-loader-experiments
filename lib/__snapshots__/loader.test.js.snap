// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should throw with circular peerDependencies 1`] = `"Probable circular references in peerDependencies. Unqualified peers are 'broken-peer-1@1.0.0', 'broken-peer-2@1.0.0'"`;

exports[`Should work 1`] = `
Object {
  "rr": Object {
    "react": "react @ 15.0.0",
    "redux": "redux @ 2.0.0",
    "stamp": "cmVhY3QtcmVkdXhANS4wLjB7cmVhY3RAMTUuMC4we30scmVkdXhAMi4wLjB9L2luZGV4Lmpz",
  },
}
`;

exports[`Should work 2`] = `
Object {
  "react": "react @ 16.0.0",
  "react-redux": Object {
    "react": "react @ 16.0.0",
    "redux": "redux @ 3.0.0",
    "stamp": "cmVhY3QtcmVkdXhANS4wLjB7cmVhY3RAMTYuMC4we30scmVkdXhAMy4wLjB9L2luZGV4Lmpz",
  },
  "react-redux-plugin": Object {
    "react-redux": Object {
      "react": "react @ 16.0.0",
      "redux": "redux @ 3.0.0",
      "stamp": "cmVhY3QtcmVkdXhANS4wLjB7cmVhY3RAMTYuMC4we30scmVkdXhAMy4wLjB9L2luZGV4Lmpz",
    },
  },
  "redux": "redux @ 3.0.0",
}
`;

exports[`Should work 3`] = `
Object {
  "react": "react @ 16.0.0",
  "react-redux": Object {
    "react": "react @ 16.0.0",
    "redux": "redux @ 2.0.0",
    "stamp": "cmVhY3QtcmVkdXhANS4wLjB7cmVhY3RAMTYuMC4we30scmVkdXhAMi4wLjB9L2luZGV4Lmpz",
  },
  "react-redux-plugin-deep": Object {
    "react-redux-plugin": Object {
      "react-redux": Object {
        "react": "react @ 16.0.0",
        "redux": "redux @ 2.0.0",
        "stamp": "cmVhY3QtcmVkdXhANS4wLjB7cmVhY3RAMTYuMC4we30scmVkdXhAMi4wLjB9L2luZGV4Lmpz",
      },
    },
  },
  "redux": "redux @ 2.0.0",
}
`;
